app/src/main/java/org/kiwix/kiwixmobile/utils/ServerUtils.java

```diff
@@ -0,0 +1,60 @@
+package org.kiwix.kiwixmobile.utils;
+
+import androidx.annotation.NonNull;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
+
+public class ServerUtils {
+  public static int port;
```

Comment:
these static variables should be done away with. The server should be the one who knows if it is running and on what port .You can fetch the instance to see what it is up to if you need to or query the instanc via HotspotService instance in the Activity. Ideally there would be observable data you connect to but that might be a bit much to ask